client/src/modules/users/UserDepotManagementModal.js

```diff
@@ -19,29 +20,97 @@ function UsersDepotManagementController($state, Users, Notify, AppCache, params)
   // the user object that is either edited or created
   vm.user = {};
   vm.depots = [];
+  vm.setNodeValue = setNodeValue;
+  vm.setAllNodeValue = setAllNodeValue;
 
   // exposed methods
   vm.submit = submit;
   vm.closeModal = closeModal;
+  vm.toggleFilter = toggleFilter;
+  vm.loading = true;
 
   vm.onDepotChange = (depots) => {
     vm.user.depots = depots;
   };
 
+  function setNodeValue(childrens, depot) {
```

Comment:
Hmm... we have a lot of the same functionality in the `bhCheckboxTree`.  I know you tried to use that already, so you don't need to do anything now.

But in a future PR, we should probably add the functionality you are looking for to the `bhCheckboxTree`.  That way, we can test the component to make sure it works correctly.

Reply:
Ok